packages/main/src/DateRangePicker.js

```diff
@@ -0,0 +1,402 @@
+			return this;
+		}
+
+		if (!value) {
+			this.value = "";
+			return;
+		}
+
+		dates = this._splitValueByDelimiter(value);
+		if (!isValid) {
+			this.valueState = ValueState.Error;
+			console.warn("Value can not be converted to a valid dates", this); // eslint-disable-line
+			return;
+		}
+		this.valueState = ValueState.None;
+
+		this._firstDateTimestamp = this.getFormat().parse(dates[0]).getTime() / 1000;
+		this._lastDateTimestamp = this.getFormat().parse(dates[1]).getTime() / 1000;
+
+		if (this._firstDateTimestamp > this._lastDateTimestamp) {
+			const temp = this._firstDateTimestamp;
+			this._firstDateTimestamp = this._lastDateTimestamp;
+			this._lastDateTimestamp = temp;
+		}
+
+		this._calendar.selectedDates = this.dateIntervalArrayBuilder(this._firstDateTimestamp * 1000, this._lastDateTimestamp * 1000);
+		this.value = this._formatValue(this._firstDateTimestamp, this._lastDateTimestamp);
+		this._prevValue = this.value;
+	}
+
+	_changeCalendarSelection(focusTimestamp) {
+		if (this._calendarDate.getYear() < 1) {
+			// 0 is a valid year, but we cannot display it
+			return;
+		}
+
+		const oCalDate = this._calendarDate,
+			timestamp = focusTimestamp || oCalDate.valueOf() / 1000,
+			dates = this._splitValueByDelimiter(this.value);
+
+		this._calendar = Object.assign({}, this._calendar);
+		this._calendar.timestamp = timestamp;
+		if (this.value && this._checkValueValidity(this.value)) {
+			this._calendar.selectedDates = this.dateIntervalArrayBuilder(this._getTimeStampFromString(dates[0]), this._getTimeStampFromString(dates[1]));
+		}
+	}
+
+	get _calendarDate() {
+		const dates = this._splitValueByDelimiter(this.value),
+			value = this._checkValueValidity(this.value) ? dates[0] : this.getFormat().format(new Date()),
+			millisecondsUTCFirstDate = value ? this.getFormat().parse(value, true).getTime() : this.getFormat().parse(this.validValue, true).getTime(),
+			oCalDateFirst = CalendarDate.fromTimestamp(
+				millisecondsUTCFirstDate - (millisecondsUTCFirstDate % (24 * 60 * 60 * 1000)),
+				this._primaryCalendarType
+			);
+
+		return oCalDateFirst;
+	}
+
+	/**
+	 * Currently selected first date represented as JavaScript Date instance.
+	 *
+	 * @readonly
+	 * @type { Date }
+	 * @public
+	 */
+	get firstDateValue() {
+		const dateValue = new Date(this._firstDateTimestamp * 1000);
+		return new Date(Date.UTC(dateValue.getFullYear(), dateValue.getMonth(), dateValue.getDate()));
+	}
+
+	/**
+	 * Currently selected last date represented as JavaScript Date instance.
+	 *
+	 * @readonly
+	 * @type { Date }
+	 * @public
+	 */
+	get lastDateValue() {
+		const dateValue = new Date(this._lastDateTimestamp * 1000);
+		return new Date(Date.UTC(dateValue.getFullYear(), dateValue.getMonth(), dateValue.getDate()));
+	}
+
+	get _placeholder() {
+		return this.placeholder !== undefined ? this.placeholder : this._displayFormat.concat(" ", this.delimiter, " ", this._displayFormat);
+	}
+
+	_handleInputChange() {
```

Comment:
Now after you refactor the DatePicker#_handleInputChange to use _checkValueValidity, you can remove this method right?

Reply:
Now we have other difference in the way we're setting the value, with the feature ( if the first written date is bigger than the second written, they should be swapped)